File: Exercises/exercise01/main.py
from threading import Thread



i = 0

def Thread_func1():
	global i
	for j in range(1000000):
		i = i + 1

		

		
def Thread_func2():
	global i
	for k in range(1000000):
		i-=1


def main():
	
	Thread1 = Thread(target = Thread_func1(), args = (),)
	Thread2 = Thread(target = Thread_func2(), args = (),)
	Thread1.start()
	Thread2.start()
    
	Thread1.join()
	Thread2.join()
	
	print(i)
	

	
main()

File: Exercises/exercise01/main.c
//gcc -std=gnu99 -Wall -g -o main main.c -lpthread



#include <pthread.h>
#include <stdio.h>



int i = 0;

void* Thread_func1(){
	int j;
	for(j = 0; j<1000000; j++){
		
		i+=1;
	}
	
	
	return NULL;
}

void* Thread_func2(){
	int j;
	for(j = 0; j<1000000; j++){
		
		i-=1;
	}
	 

	return NULL;
}



int main(){
	
    pthread_t Thread1, Thread2;
    pthread_create(&Thread1, NULL, Thread_func1, NULL);
    pthread_create(&Thread2, NULL, Thread_func2, NULL);
    pthread_join(Thread1, NULL);
    pthread_join(Thread2, NULL);
    printf("Value of i: %d \n", i);
	
	return 0;
}
